src/sentry/rules/processing/delayed_processing.py

```diff
@@ -200,6 +202,31 @@ def parse_rulegroup_to_event_data(
     return parsed_rulegroup_to_event_data
 
 
+def bulk_fetch_events(event_ids: list[str], project_id: int) -> set[Event]:
+    node_id_to_event_id = {
+        Event.generate_node_id(project_id, event_id=event_id): event_id for event_id in event_ids
+    }
+
+    try:
+        bulk_data = nodestore.get_multi(list(node_id_to_event_id.keys()))
+    except (ServiceUnavailable, DeadlineExceeded):
+        try:
+            bulk_data = nodestore.get_multi(list(node_id_to_event_id.keys()))
+        except (ServiceUnavailable, DeadlineExceeded):
+            bulk_data = nodestore.get_multi(list(node_id_to_event_id.keys()))
```

Comment:
It'd be cool if this just called bulk_fetch_events again instead of repeating the code.